acnportal/acnsim/models/battery.py

```diff
@@ -156,13 +156,13 @@ def charge(self, pilot, voltage, period):
         if self._soc < self._transition_soc:
             charge_power = min([pilot * voltage / 1000, self._max_power, rate_to_full])
             if self._noise_level > 0:
-                charge_power -= abs(np.random.normal(0, self._noise_level))
+                charge_power = max(charge_power - abs(np.random.normal(0, self._noise_level)), 0)
         else:
             charge_power = min([pilot * voltage / 1000,
                                 (1 - self._soc) / (1 - self._transition_soc) * self._max_power,
                                 rate_to_full])
             if self._noise_level > 0:
-                charge_power += np.random.normal(0, self._noise_level)
+                charge_power = min(max(charge_power + np.random.normal(0, self._noise_level), 0), pilot * voltage / 1000, self._max_power, rate_to_full)
```

Comment:
For clarify, perhaps we should use numpy.clip (https://numpy.org/doc/stable/reference/generated/numpy.clip.html?highlight=clip#numpy.clip)

For example we could say:
```suggestion
                upper_bound = min(pilot * voltage / 1000, self._max_power, rate_to_full)
                charge_power += np.random.normal(0, self._noise_level)
                charge_power = np.clip(charge_power, 0, upper_bound)
```

Reply:
We should also probably add some tests in tests/test_battery.py, to make sure this works as we expect.